lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+					<div
+						v-if="footerTertiaryButtonText || footerCheckboxText"
+						class="modal__footerColumn --ctaSecondary"
+					>
+						<div v-if="footerCheckboxText" ref="button" class="modal__checkbox">
+							<input
+								id="modal__checkboxInput"
+								type="checkbox"
+								:checked="false"
+								class="modal__checkboxInput"
+								@change="$emit('checkbox-change')"
+							/>
+							<label for="modal__checkboxInput" class="modal__checkboxLabel">
+								{{ footerCheckboxText }}
+							</label>
+						</div>
+						<wnl-button
+							v-if="footerTertiaryButtonText"
+							ref="button"
+							class="modal__buttonSecondary"
+							:type="BUTTON_TYPES.TEXT"
+							:color="BUTTON_COLORS.NEUTRAL"
+							:icon-left="ICONS[footerTertiaryButtonIcon]"
+						>
+							{{ footerTertiaryButtonText }}
+						</wnl-button>
+					</div>
+					<div
+						v-if="footerSecondaryButtonText || footerPrimaryButtonText"
+						class="modal__footerColumn --cta"
+					>
+						<wnl-button
+							v-if="footerSecondaryButtonText"
+							ref="button"
+							class="modal__buttonSecondary"
+							:type="BUTTON_TYPES.OUTLINED"
+							:color="calcFooterSecondaryButtonColor"
+							:icon-right="ICONS[footerSecondaryButtonIcon]"
+						>
+							{{ footerSecondaryButtonText }}
+						</wnl-button>
+						<wnl-button
+							v-if="footerPrimaryButtonText"
+							ref="button"
+							class="modal__buttonPrimary"
+							:color="calcFooterPrimaryButtonColor"
+							:icon-right="ICONS[footerPrimaryButtonIcon]"
+						>
+							{{ footerPrimaryButtonText }}
+						</wnl-button>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/shadows';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/icons';
+@import '../../../../styles/settings/media-queries';
+
+$modal-medium-width: 700px;
+$modal-small-width: 460px;
+$image-height: 200px;
+$image-height-small: 140px;
+
+.modal {
+	$self: &;
+
+	background: $color-default-overlay;
+	padding: $space-l $space-s;
+
+	@media #{breakpoint-s()} {
+		padding: $space-l;
+	}
+
+	&__wrapper {
+		background: $color-default-background;
+		border-radius: $radius-m;
+		box-shadow: $shadow-xl;
+		margin: 0 auto;
+		overflow: hidden;
+		position: relative;
+		max-width: $modal-medium-width;
+
+		&.-small {
+			max-width: $modal-small-width;
+
+			#{$self}__image {
+				height: $image-height-small;
+			}
+			#{$self}__content {
+				padding-left: $space-m;
+				padding-right: $space-m;
+			}
+		}
+	}
+
+	&__content {
+		padding: $space-l $space-s;
+
+		@media #{breakpoint-s()} {
+			padding: $space-l $space-xl;
+		}
+
+		&.-centered {
+			#{$self}__header,
+			#{$self}__textContent {
+				text-align: center;
```

Comment:
does it work with content other than text? Icons or other components?

Reply:
With simple component like Icon it will works. I asked UX about this center, because `<ul>` will looks ugly. UX said the DialogModal with centered will be used only for simple cases like text only.